infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -156,9 +156,11 @@ private boolean applyMsg(LogReplicationEntryMsg txMessage) {
                             if (streamId.equals(REGISTRY_TABLE_ID)) {
                                 // If registry table entries are being handled, indicate the config to sync with
                                 // registry table after this transaction.
-                                registryTableUpdated.set(true);
                                 smrEntries = filterRegistryTableEntries(new ArrayList<>(smrEntries));
-                                log.info("Registry Table entries during log entry sync = {}", smrEntries.size());
+                                if (!smrEntries.isEmpty()) {
+                                    log.info("Registry Table entries during log entry sync = {}", smrEntries.size());
+                                    registryTableUpdated.set(true);
+                                }
```

Comment:
I agree with this fix, and I think previously we removed this check for a corner case when:
* We didn't add the sync at the beginning (the one for long duration between snapshot/log entry sync and another log entry sync)
* Streams to drop was checked with registry table directly

it will cause the "no new streams" issue at that time. As now we changed the places above , I agree it's fine to add this check back to avoid unnecessary sync

thanks!